x-pack/solutions/security/plugins/security_solution/public/common/hooks/timeline/use_timeline_api.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import type { Filter } from '@kbn/es-query';
+import { useCreateTimeline } from '../../../timelines/hooks/use_create_timeline';
+import { updateProviders, setFilters } from '../../../timelines/store/actions';
+import { SourcererScopeName } from '../../../sourcerer/store/model';
+import type { DataProvider } from '../../../../common/types';
+import { sourcererSelectors } from '../../store';
+import { sourcererActions } from '../../store/actions';
+import { inputsActions } from '../../store/inputs';
+import { InputsModelId } from '../../store/inputs/constants';
+import type { TimeRange } from '../../store/inputs/model';
+import { TimelineId } from '../../../../common/types/timeline';
+import { TimelineTypeEnum } from '../../../../common/api/timeline';
+
+interface OpenTimelineArgs {
+  /**
+   * The data providers to apply to the timeline.
+   */
+  dataProviders?: DataProvider[] | null;
+
+  /**
+   * The filters to apply to the timeline.
+   */
+  filters?: Filter[] | null;
+
+  /**
+   * The time range to apply to the timeline, defaults to global time range.
+   */
+  timeRange?: TimeRange;
+
+  /**
+   * Whether to keep the current data view or reset it to the default.
+   */
+  keepDataView?: boolean;
+}
+
+export const useTimelineApi = () => {
+  const dispatch = useDispatch();
+
+  const signalIndexName = useSelector(sourcererSelectors.signalIndexName);
+  const defaultDataView = useSelector(sourcererSelectors.defaultDataView);
+
+  const clearTimelineTemplate = useCreateTimeline({
+    timelineId: TimelineId.active,
+    timelineType: TimelineTypeEnum.template,
+  });
+
+  const clearTimelineDefault = useCreateTimeline({
+    timelineId: TimelineId.active,
+    timelineType: TimelineTypeEnum.default,
+  });
+
+  const openTimeline = useCallback(
```

Comment:
if you're ok renaming the `useInvestigateInTimeline` hook, I think this could become `investigateInTimeline` instead of `openTimeline`. This will make it more obvious that internally it will clear the state of the old timeline.